db/db_impl/db_impl.h

```diff
@@ -2963,6 +2966,10 @@ class DBImpl : public DB {
   // stores the number of compactions are currently running
   int num_running_compactions_;
 
+  // stores the number of input iterators required for currently running
+  // compactions
+  int num_running_compaction_input_iterators_;
```

Comment:
I also would have expected to use something like `InstrumentedMutexLock l(&mutex_);` or at least `std::std::atomic<int> ` to protect `num_running_compactions_`. 

Again, I followed the pattern here, but maybe we should also be updating `num_running_compactions_` too? 

For instance, I see `num_running_flushes_` gets surrounded by `InstrumentedMutexLock l(&mutex_)`